compiler/src/dotty/tools/dotc/inlines/Inlines.scala

```diff
@@ -49,7 +49,13 @@ object Inlines:
 
   /** Can a call to method `meth` be inlined? */
   def isInlineable(meth: Symbol)(using Context): Boolean =
-    meth.is(Inline) && meth.hasAnnotation(defn.BodyAnnot) && !inInlineMethod
+    meth.is(Inline)
+    && (meth.hasAnnotation(defn.BodyAnnot)
+        // Ensure `isInlineable` works with temporary symbols created during
+        // overloading resolution by `Applications#resolveMapped`.
+        // Testcase: tests/pos/i21410c.scala
+        || meth.hasAnnotation(defn.MappedAlternativeAnnot))
```

Comment:
Does being mapped imply `hasAnnotation(defn.BodyAnnot)`?  Why not make the symbol copying in `resolveMapped` copy the annotations too?  Or (and I'm not sure if this is possible) add a backreference from the `MappedAlternativeAnnot` symbol to the original?